src/ip.rs

```diff
@@ -71,20 +71,20 @@ impl Ipv4Hdr {
 
 #[cfg(feature = "std")]
 impl Ipv4Hdr {
-    pub fn src(&self) -> std::net::Ipv4Addr {
-        std::net::Ipv4Addr::from(self.src_addr)
+    pub fn src_addr(&self) -> std::net::Ipv4Addr {
```

Comment:
It would be nice to add a docstring (in this and all the other methods) mentioning that the endianness is being converted here. Maybe we should even somehow include that information in the method name itself (e.g. `src_addr_from_be`).

In general, we didn't use to convert endianness in this library. If we do, I think we should warn the user about that.